pkg/infradriver/ssh_infradriver.go

```diff
@@ -181,17 +196,47 @@ func (d *SSHInfraDriver) GetHostEnv(host net.IP) map[string]interface{} {
 }
 
 func (d *SSHInfraDriver) GetClusterEnv() map[string]interface{} {
-	// Set the default RegistryDomain and RegistryPort env,
-	// and override the env if the user specifies RegistryDomain and RegistryPort env
+	// expose RegistryDomain and RegistryPort to env in order others needed.
+	regConfig := d.GetClusterRegistryConfig()
 	if _, ok := d.clusterEnv["RegistryDomain"]; !ok {
-		d.clusterEnv["RegistryDomain"] = common.DefaultRegistryDomain
+		d.clusterEnv["RegistryDomain"] = regConfig.LocalRegistry.Domain
 	}
 	if _, ok := d.clusterEnv["RegistryPort"]; !ok {
-		d.clusterEnv["RegistryPort"] = common.DefaultRegistryPort
+		d.clusterEnv["RegistryPort"] = regConfig.LocalRegistry.Port
 	}
+
 	return d.clusterEnv
 }
 
+func (d *SSHInfraDriver) GetClusterRegistryConfig() v2.Registry {
+	clusterRegistryConfig := d.cluster.Spec.Registry
+	// configure default cluster registry
+	if clusterRegistryConfig.LocalRegistry == nil {
```

Comment:
i think maybe we can add a flag named HA for LocalRegistry, if true, will set deployHosts as all masters.

Reply:
Now, user need manually specify deployhosts, which is incomprehensible and troublesome